crates/wasmtime/src/coredump.rs

```diff
@@ -80,6 +83,194 @@ impl WasmCoreDump {
     pub fn memories(&self) -> &[Memory] {
         self.memories.as_ref()
     }
+
+    /// Serialize this core dump into [the standard core dump binary
+    /// format][spec].
+    ///
+    /// Once serialized, you can write this core dump do disk, send it over the
+    /// network, or pass it to other debugging tools that consume Wasm core
+    /// dumps.
+    ///
+    /// [spec]: https://github.com/WebAssembly/tool-conventions/blob/main/Coredump.md
+    pub fn serialize(&self, mut store: impl AsContextMut, name: &str) -> Vec<u8> {
+        let store = store.as_context_mut();
+        self._serialize(store, name)
+    }
+
+    fn _serialize<T>(&self, mut store: StoreContextMut<'_, T>, name: &str) -> Vec<u8> {
```

Comment:
If you're feeling sufficiently motivated I think the `T` here could be dropped in theory  but would require some internal refactoring as it seems like it's required or calling public APIs which don't have private equivalents working with `StoreOpaque`

Reply:
> it seems like it's required or calling public APIs which don't have private equivalents working with `StoreOpaque`

Yeah, this is why the `T` is propagated. A bit unfortunate, but also something I don't want to deal with at this very moment. I can file a follow up issue.

Reply:
https://github.com/bytecodealliance/wasmtime/issues/7083